Code/Tools/AssetProcessor/native/AssetManager/AssetRequestHandler.cpp

```diff
@@ -107,6 +107,46 @@ namespace
                     auto resultMove = relocationInterface->Move(
                         messageData.m_message->m_fromPath, messageData.m_message->m_toPath, false, true, true, true);
 
+                    if (resultMove.IsSuccess())
+                    {
+                        AssetProcessor::RelocationSuccess success = resultMove.TakeValue();
+
+                        // The report can be too long for the AZ_Printf buffer, so split it into individual lines
+                        AZStd::string report =
+                            relocationInterface->BuildChangeReport(success.m_relocationContainer, success.m_updateTasks);
+                        AzFramework::StringFunc::Tokenize(report.c_str(), lines, "\n");
+
+                        for (const AZStd::string& line : lines)
+                        {
+                            AZ_Printf(AssetProcessor::ConsoleChannel, (line + "\n").c_str());
+                        }
+                    }
+                    break;
+                }
+                case AssetChangeReportRequest::ChangeType::CheckDelete:
+                {
+                    auto resultCheck = relocationInterface->Delete(messageData.m_message->m_fromPath);
+
+                    if (resultCheck.IsSuccess())
```

Comment:
could you please factor out a little lambda or member function that takes a result and does this report string / console print and then call it from line 110 and 130 here, so there's no duplicated code to maintain?

Reply:
Implemented as a member function.